include/nccl_ofi_freelist.h

```diff
@@ -201,36 +164,10 @@ static inline void nccl_ofi_freelist_entry_set_undefined(nccl_ofi_freelist_t *fr
 {
 	size_t user_entry_size = freelist->entry_size - MEMCHECK_REDZONE_SIZE;
 
-	if (freelist->have_reginfo) {
-		size_t reginfo_offset = freelist->reginfo_offset;
-		size_t elem_size = sizeof(struct nccl_ofi_freelist_elem_t);
-		size_t reginfo_size = sizeof(struct nccl_ofi_freelist_reginfo_t);
-		size_t redzone_offset = offsetof(struct nccl_ofi_freelist_reginfo_t, redzone);
-
-		/* Entry after reginfo structure is accessible but undefined */
-		nccl_net_ofi_mem_undefined_unaligned((void*)((uintptr_t)entry_p + reginfo_offset + reginfo_size),
-						     user_entry_size - reginfo_offset - reginfo_size);
-		/* Redzone at the end of the reginfo structure is
-		 * marked as not accessible */
-		nccl_net_ofi_mem_noaccess_unaligned((void*)((uintptr_t)entry_p + reginfo_offset + redzone_offset),
-						    MEMCHECK_REDZONE_SIZE);
-		/* Members of reginfo structure except first and last
-		 * member are accessible and defined */
-		nccl_net_ofi_mem_defined_unaligned((void*)((uintptr_t)entry_p + reginfo_offset + elem_size),
-						   redzone_offset - elem_size);
-		/* First member of reginfo structure, i.e.,
-		 * nccl_ofi_freelist_elem_t structure, is marked as
-		 * not accessible */
-		nccl_net_ofi_mem_noaccess_unaligned((void*)((uintptr_t)entry_p + reginfo_offset), elem_size);
-		/* First part of entry until reginfo structure is
-		 * accessible but undefined */
-		nccl_net_ofi_mem_undefined(entry_p, reginfo_offset);
-	} else {
-		/* Entry allocated by the user is accessible but
-		 * undefined. Note that this allows the user to
-		 * override the nccl_ofi_freelist_elem_t structure. */
-		nccl_net_ofi_mem_undefined(entry_p, user_entry_size);
-	}
+	/* Entry allocated by the user is accessible but
+	 * undefined. Note that this allows the user to
+	 * override the nccl_ofi_freelist_elem_t structure. */
+	nccl_net_ofi_mem_undefined(entry_p, user_entry_size);
```

Comment:
had an offline question about whether this was doing the right thing for nonzero padding/redzone size, addressed over slack, thanks